File: src/main/java/net/flyingfishflash/loremlist/api/data/request/LrmListItemAddRequest.java
package net.flyingfishflash.loremlist.api.data.request;

import java.util.Set;
import java.util.UUID;
import net.flyingfishflash.loremlist.core.validation.ValidUuidSet;

public record LrmListItemAddRequest(@ValidUuidSet(allowEmpty = false) Set<UUID> itemIdCollection) {}

File: src/main/kotlin/net/flyingfishflash/loremlist/api/data/request/LrmListItemAddRequest.kt
package net.flyingfishflash.loremlist.api.data.request

import kotlinx.serialization.Serializable
import net.flyingfishflash.loremlist.core.serialization.UuidSerializer
import net.flyingfishflash.loremlist.core.validation.ValidUuidSet
import java.util.*

@Serializable
data class LrmListItemAddRequest(
  @field:ValidUuidSet(allowEmpty = false)
  val itemIdCollection: Set<
    @Serializable(UuidSerializer::class)
    UUID,
    >,
)
